spcr-settings: fix finding the advanced settings button and allow passing in settings in constructor

jeroentvb opened the pull request:
This PR does the following:
* Fix finding the settings button in a non english UI (code was looking for some english text, but my spotify is set to Dutch)
* Allow passing in settings into the constructor. In my opinion calling a function for every setting item was a bit cumbersome.
* Add `keyDown` and `blur` events to the input field
* Refactor some `for` loops to es6 array functions

jeroentvb committed:
refactor(spcr-settings): improve some code

jeroentvb committed:
fix(spcr-settings): finding advanced settings button in on english UI

jeroentvb committed:
feat(spcr-settings): allow passing in settings in the constructor

jeroentvb committed:
feat(spcr-settings): add keyDown and blur event emitters to input field

FlafyDev commented:
Thanks for the PR! I haven't thought about instances where the settings button is not in English so thanks for fixing it.
I'll only be able to test your changes in a few hours but for now I looked at the code and I'm not sure why you added a `keyDown` event instead of the already implemented `onChange` event.

jeroentvb commented:
I've implemented the `keyDown` event because `onChange` only emits the current value while I need the `keyboardEvent` to register pressed keys in the power-bar. See my code [here](https://github.com/jeroentvb/spicetify-power-bar/blob/41f67d925a04d57402a92991ab2b723acf5f82b6/src/components/PowerBar.tsx#L186-L221).

FlafyDev commented:
Well in that case it may be better to have a way to add your own attributes to the input element instead of adding every attribute to the `ISettingsField`.

jeroentvb commented:
Good point, and I agree. I'll change it.

jeroentvb committed:
refactor(spcr-settings): allow passing in custom input event listeners, and add settings field union type

jeroentvb commented:
I've implemented adding your own attributes to the input field, and as a bonus added a union type for the settings field.
Based on the set `type` of settings field you'll get better typing and errors for missing fields, or having non-existent fields.

FlafyDev committed:
refactor(spcr-settings): more custom fields.
Added custom events for every field.
The 'callback' properties are now gone and are considered a custom event
Renamed GenericToggle to Toggle.

FlafyDev committed:
refactor(spcr-settings): Reverted automatic key names.
Not sure how it's called.
I'm not used to it and I personally think it looks better how it was before...

FlafyDev commented:
Awesome! The custom events are very useful so I expanded them to the other fields as well. I think I'll do more testing tomorrow and then merge.

jeroentvb commented:
Great stuff!

I also think the release with these changes should be a major version bump, as they are breaking. Unless the package is still in beta.
